articles/ml-meets-swe.md

```diff
@@ -0,0 +1,29 @@
+
+1.	Software developers encode domain knowledge into explicit, executable rules, such as if-then statements. ML practitioners on the other hand take a sufficient number of input data examples and attach the intended results as so-called labels. Hence: Labelling is the new programming.
+
+2.	Software developers compile their software artefacts to create executable binaries. ML Practitioners automatically adjust the thousands or even millions of parameters of some chosen standard algorithm until that algorithm returns the intended output for their labelled input data. This so-called training procedure is the new compilation.
+
+3.	Software developers (hopefully) write a lot of tests to prove correctness and provide a stability harness for their code. ML practitioners closely observe key statistical properties of their input and output data.
+
+It should be obvious in the above, though, that we intentionally ignore the not so subtle differences that are still relevant, if you want to understand the whole picture. For example, machine learning cannot (yet) be used to devise or even create attractive and effective user interfaces; and rule-based systems to e.g. transcribe human speech cannot catch up with the performance of modern neural networks anymore. So, there are differences. There is room for both professional practises. 
+
+And to meet the modern consumers’ ever-growing appetite for smart applications, practitioners of both fields need to unite. Unfortunately, that come-together sometimes turns out to be not as easy as the naive observer might expect. Indeed, if you think back, then the difficulties should not even come as a surprise. If you are a seasoned developer, you may remember hearing sentences like “I’m a software developer, not a database admin” or “I’m a software developer, not a system operator”. If you are new to the field, though, you may not have heard any such things, and the reason is probably that there is already a cure for this problem. It’s called DevOps; and it appears to be a very satisfactory common field for all participants, that's probably why they met there in peace. DevOps effectively managed to unite software engineering and the classical related fields. 
+
+Nowadays you may hear “I’m a data scientist, not a software engineer”. While that may be perfectly accurate, the need to emphasize the fact unfortunately implies the claim: “I don’t need to care what it takes to use my results in production”. On the other side, typical enterprise developers faced with the need to understand, write or simply integrate Python code find it hard to overcome their acquired resentment for scripting languages. 
```

Comment:
I don’t need to care what it takes to use my results in production -> stimmt die Unterstellung wirklich?